### PR TITLE
Issue generate graph

### PR DESCRIPTION
**Pull request by nazzaninghazzi**
## Motivation and Context

A new feature for generating graphs based on user input courses was added recently. However, clicking the 'Generate Graph' button faced an error. This pull request added changes to the code to fix the error. 

When clicking the 'Generate Graph' button, some errors regarding wrong variable types and accessing properties of undefined variables were generated in the console. For example, arrays were passed to `Graph` states instead of objects. Although some of the states of the graph such as `nodesStatus`, `BoolsStatus`, and `edgesStatus` were not defined, the program accessed their properties.

## Your Changes

**Description**:
Changed the type of the variables from array to objects to be passed to `Graph` states. These variables included `nodesJSON`, `edgesJSON`, `hybridsJSON`, and `boolsJSON`. These objects map from the component id to the component.
Added new variables of type object such as `nodesStatus` to fix the issue of undefined 'state'. Added `edgesStatus` and `boolsStatus` to render and update the status of edges and bools correctly after selecting, unselecting, or hovering over the nodes.
Finally passed these new variables to the states of the `Graph`.

**Type of change** (select all that apply):

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

Input one or more example courses and make sure that after clicking the 'Generate Graph' button, the website doesn't crash and produces the expected graph.
## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CircleCI tests have passed.

### CODE REVIEW

**Review by david-yz-liu — COMMENTED** [review]
@nazzaninghazzi good work! I left a couple of comments

File: js/components/generate/generate.js [hunk 1]
```diff
@@ -122,15 +127,26 @@ class GenerateForm extends React.Component {
...
+            selected: ["active", "overridden"].indexOf(state) >= 0,
+          }
         })
 
-        hybridsList.forEach(hybrid => {
+        Object.values(hybridsJSON).forEach(hybrid => {
           childrenObj[hybrid.id_] = []
           outEdgesObj[hybrid.id_] = []
+          const nodesList = Object.values(nodesJSON)
           populateHybridRelatives(hybrid, nodesList, parentsObj, childrenObj)
```
[the remark this answers is not in the document]
Comment (nazzaninghazzi) [reply js/components/generate/generate.js:141]: What do you exactly mean by `pouplateHybridRelatives` changed?

File: app/WebParsing/ParseAll.hs [hunk 2]
```diff
@@ -2,10 +2,10 @@ module WebParsing.ParseAll
      (parseAll) where
 
 import WebParsing.ArtSciParser (parseArtSci, parseBuildings)
--- import WebParsing.UtsgJsonParser (getAllCourses)
```
Comment (david-yz-liu) [inline app/WebParsing/ParseAll.hs:5]: Please revert the changes in this file (not related to this PR)

File: js/components/generate/generate.js [hunk 3]
```diff
@@ -79,58 +79,81 @@ class GenerateForm extends React.Component {
...
         const outEdgesObj = {}
         const storedNodes = new Set()
 
-        const labelsList = data.texts.filter(function (entry) {
-          // filter for mark percentages, allow preceding characters for potential geq
-          return entry.text.match(/.*[0-9]*%/g)
+        data.texts.forEach(entry => {
+          if (entry.text.match(/.*[0-9]*%/g)) {
+            labelsJSON[entry.rId] = entry
+            // filter for mark percentages, allow preceding characters for potential geq
```
Comment (david-yz-liu) [inline js/components/generate/generate.js:99]: This is good, but the conventional placement would be to put it above the if condition

**Review by david-yz-liu — COMMENTED** [review]
@nazzaninghazzi good work! I just left a few small cleanup comments

**Review by david-yz-liu — APPROVED** [review]
